src/promptflow/promptflow/executor/_async_nodes_scheduler.py

```diff
@@ -61,12 +80,14 @@ async def _execute_with_thread_pool(
         context: FlowExecutionContext,
     ) -> Tuple[dict, dict]:
         flow_logger.info(f"Start to run {len(nodes)} nodes with the current event loop.")
+        global ASYNC_DAG_MANAGER_COMPLETED
```

Comment:
Is it possible to avoid global variable?

Reply:
Changed to Event